tests/integration/test_upgrade_contract.py

```
@@ -0,0 +1,805 @@
+
+        # Verify upgrade worked
+        version = call_contract_method(address, "get_version")
+        assert version == "v2"
+
+        # Nonce should have incremented
+        nonce_after = rpc_call("gen_getContractNonce", [address])
+        assert nonce_after > nonce_before, f"Nonce should increment: {nonce_before} -> {nonce_after}"
+
+        # Now upgrade to V3
+        nonce_bytes_v3 = nonce_after.to_bytes(32, byteorder="big")
+        code_hash_v3 = Web3.keccak(text=CONTRACT_V3)
```

Ruff (0.14.8)

Undefined name `CONTRACT_V3`

(F821)